File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventsByCategoryAndStatusQueryResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * List of events of a specified category.
 */
open class EventsByCategoryAndStatusQueryResponse : GraphQLResult<MutableMap<String, List<Event?>>>() {

    companion object {
        const val OPERATION_NAME: String = "eventsByCategoryAndStatus"
    }

    /**
     * List of events of a specified category.
     */
    fun eventsByCategoryAndStatus(): List<Event?> {
        val data: MutableMap<String, List<Event?>> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}

File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventsByCategoryAndStatusQueryResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * List of events of a specified category.
 */
class EventsByCategoryAndStatusQueryResponse extends GraphQLResult[JMap[String, scala.Seq[Event]]] {

    /**
     * List of events of a specified category.
     */
    def eventsByCategoryAndStatus(): scala.Seq[Event] = {
        val data: JMap[String, scala.Seq[Event]] = getData
        if (data != null) data.get(EventsByCategoryAndStatusQueryResponse.OPERATION_NAME) else null
    }

}

object EventsByCategoryAndStatusQueryResponse {

    private final val OPERATION_NAME: String = "eventsByCategoryAndStatus"

}
